### PR TITLE
Simplify consistent checks in class Mapping

### PR DESCRIPTION
**Pull request by Captain1653**
Simplify consistent checks in class Mapping

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java [hunk 1]
```diff
@@ -66,14 +67,12 @@ public TypeSelector(TypeFactory typeFactory, FormattingMessager messager) {
             List<List<ParameterBinding>> parameterBindingPermutations =
                 getCandidateParameterBindingPermutations( availableBindings, method.getMethod().getParameters() );
 
-            if ( parameterBindingPermutations != null ) {
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java]: This makes sense from a programming perspective. However, in our case `getMatchingParameterBinding` is quite a complex method that would execute complex logic based on `parameterBindingPermutations`. Therefore I would revert the changes to the `TypeSelector` class.

Comment (Captain1653) [reply processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java]: Ok

**Comment by Captain1653** [comment]
> Thanks for the PR @Captain1653. I left a comment about the change in the `TypeSelector`. In my opinion we shouldn't be changing that one

I have chanched
